hq_superset/views.py

```diff
@@ -236,3 +240,48 @@ def select(self, hq_domain):
         response.set_cookie('hq_domain', hq_domain)
         DomainSyncUtil(superset.appbuilder.sm).sync_domain_role(hq_domain)
         return response
+
+
+class DataSetChangeAPI(BaseSupersetView):
+    """
+    Accepts changes to datasets from CommCare HQ data forwarding
+    """
+
+    MAX_REQUEST_LENGTH = 10_485_760  # reject >10MB JSON requests
+
+    def __init__(self):
+        self.route_base = '/hq_webhook'
+        self.default_view = 'post'
+        super().__init__()
+
+    @expose('/change/', methods=['POST'])
+    # TODO: Authenticate
+    def post(self) -> FlaskResponse:
+        if request.content_length > self.MAX_REQUEST_LENGTH:
+            return json_error_response(
+                HTTPStatus.REQUEST_ENTITY_TOO_LARGE.description,
+                status=HTTPStatus.REQUEST_ENTITY_TOO_LARGE.value,
+            )
+
+        try:
+            request_json = json.loads(request.get_data(as_text=True))
+        except json.JSONDecodeError:
+            return json_error_response(
+                'Invalid JSON syntax',
+                status=HTTPStatus.BAD_REQUEST.value,
+            )
+
+        try:
+            change = DataSetChange(**request_json)
```

Comment:
Early suggestion: What if you move lines 275 and 284-287 into the first try on line 266 (so the happy case is inside the try) and then handle the "sad" case in the rest of the function?